sherpa-onnx/csrc/sherpa-onnx-offline-parallel.cc

```diff
@@ -0,0 +1,292 @@
+// sherpa-onnx/csrc/sherpa-onnx-offline.cc
+//
+// Copyright (c)  2022-2023  Xiaomi Corporation
+
+#include <stdio.h>
+
+#include <atomic>
+#include <chrono>  // NOLINT
+#include <fstream>
+#include <mutex>  // NOLINT
+#include <string>
+#include <thread>  // NOLINT
+#include <vector>
+
+#include "sherpa-onnx/csrc/offline-recognizer.h"
+#include "sherpa-onnx/csrc/parse-options.h"
+#include "sherpa-onnx/csrc/wave-reader.h"
+
+std::atomic<int> wav_index(0);
+std::mutex mtx;
+
+std::vector<std::vector<std::string>> split_to_batchsize(
+    const std::vector<std::string> &input, int32_t batch_size) {
+  std::vector<std::vector<std::string>> outputs;
+  auto itr = input.cbegin();
+  int32_t process_num = 0;
+
+  while (process_num + batch_size <= static_cast<int32_t>(input.size())) {
+    auto chunk_end = itr + batch_size;
+    outputs.emplace_back(itr, chunk_end);
+    itr = chunk_end;
+    process_num += batch_size;
+  }
+  if (itr != input.cend()) {
+     outputs.emplace_back(itr, input.cend());
+  }
+  return outputs;
+}
+
+std::vector<std::string> load_scp_file(const std::string &wav_scp_path) {
+  std::vector<std::string> wav_paths;
+  std::ifstream in(wav_scp_path);
+  if (!in.is_open()) {
+      fprintf(stderr, "Failed to open file: %s.\n", wav_scp_path.c_str());
+      return wav_paths;
```

Comment:
```suggestion
      exit(-1);
```

We can just exit on errors.

Reply:
when return a empty vector, the main function will exit normally.  may be it's a better way?

Reply:
ok, it's up to you.